# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/style/nsTransitionManager.h
# Commit: 63603267402e
# Full Hash: 63603267402ee93d6551ef60c35191d99aaee06b
# Author: Zach Hoffman <zach@zrhoffman.net>
# Date: 2023-12-18 17:17:57
# Regressor Bug: 1846516
# File Overlap Count: 3
# Description:
#   Bug 1846516 - [css-properties-values-api] Introduce AnimatedPropertyID/AnimatedPropertIDSet. r=firefox-animation-reviewers,firefox-style-system-reviewers,layout-reviewers,emilio,hiro
#   
#   This will make possible to animate custom properties on Gecko side. For now, the
#   animation code keeps only dealing with nsCSSPropertyID, so behavior is unchanged.
#   
# ==============================================================================

diff -r 753bc39f59ff -r 63603267402e layout/style/nsTransitionManager.h
--- a/layout/style/nsTransitionManager.h	Mon Dec 18 09:24:07 2023 +0000
+++ b/layout/style/nsTransitionManager.h	Mon Dec 18 09:24:08 2023 +0000
@@ -18,6 +18,7 @@
 struct nsStyleUIReset;
 
 namespace mozilla {
+class AnimatedPropertyIDSet;
 class ComputedStyle;
 enum class PseudoStyleType : uint8_t;
 }  // namespace mozilla
@@ -60,17 +61,17 @@
 
   // Returns whether the transition actually started.
   bool ConsiderInitiatingTransition(
-      nsCSSPropertyID aProperty, const nsStyleUIReset& aStyle,
+      const mozilla::AnimatedPropertyID&, const nsStyleUIReset& aStyle,
       uint32_t transitionIdx, mozilla::dom::Element* aElement,
       mozilla::PseudoStyleType aPseudoType,
       CSSTransitionCollection*& aElementTransitions,
       const mozilla::ComputedStyle& aOldStyle,
       const mozilla::ComputedStyle& aNewStyle,
-      nsCSSPropertyIDSet& aPropertiesChecked);
+      mozilla::AnimatedPropertyIDSet& aPropertiesChecked);
 
   already_AddRefed<mozilla::dom::CSSTransition> DoCreateTransition(
-      nsCSSPropertyID aProperty, mozilla::dom::Element* aElement,
-      mozilla::PseudoStyleType aPseudoType,
+      const mozilla::AnimatedPropertyID& aProperty,
+      mozilla::dom::Element* aElement, mozilla::PseudoStyleType aPseudoType,
       const mozilla::ComputedStyle& aNewStyle,
       CSSTransitionCollection*& aElementTransitions,
       mozilla::TimingParams&& aTiming, mozilla::AnimationValue&& aStartValue,
